src/utils.jl

```diff
@@ -388,6 +388,32 @@ Remove the units from a quantity.
 ustrip(::AbstractDimensions) = error("Cannot remove units from an `AbstractDimensions` object.")
 @inline ustrip(q) = q
 
+"""
+    ustrip(unit::UnionAbstractQuantity, q::UnionAbstractQuantity)
+
+Convert quantity `q` to the units specified by `unit`, then strip the units.
+This is equivalent to `ustrip(q / unit)`, but also verifies the dimensions are compatible.
+
+# Examples
+```julia
+julia> ustrip(u"km", 1000u"m")
+1.0
+
+julia> ustrip(u"s", 1u"minute")
+60.0
+
+julia> ustrip(u"km", [1000u"m", 2000u"m"])
+2-element Vector{Float64}:
+ 1.0
+ 2.0
+```
+"""
+@inline function ustrip(unit::UnionAbstractQuantity, q::UnionAbstractQuantity)
+    unit, q = promote_except_value(unit, q)
+    dimension(unit) == dimension(q) || throw(DimensionError(unit, q))
+    return ustrip(q / unit)
```

Comment:
Since we've already verified dimensions, I _think_ we can do

```julia
    return ustrip(q) / ustrip(unit)
```

which is faster